packages/toolpad-app/src/components/propertyControls/SelectOptions.tsx

```diff
@@ -0,0 +1,226 @@
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  FormControl,
+  IconButton,
+  InputLabel,
+  OutlinedInput,
+  List,
+  ListItem,
+  ListItemButton,
+  ListItemText,
+  FormHelperText,
+  Stack,
+  TextField,
+} from '@mui/material';
+import * as React from 'react';
+import DeleteIcon from '@mui/icons-material/Delete';
+import ArrowBackIcon from '@mui/icons-material/ArrowBack';
+import type { EditorProps } from '../../types';
+
+interface SelectOption {
+  label?: string;
+  value: string;
+}
+
+function SelectOptionsPropEditor({
+  label,
+  value = [],
+  onChange,
+}: EditorProps<(string | SelectOption)[]>) {
+  const [editOptionsDialogOpen, setEditOptionsDialogOpen] = React.useState(false);
+  const optionInputRef = React.useRef<HTMLInputElement | null>(null);
+  const [editingIndex, setEditingIndex] = React.useState<number | null>(null);
+
+  const editingOption: SelectOption | null = React.useMemo(() => {
+    if (typeof editingIndex === 'number') {
+      const option: SelectOption | string = value[editingIndex];
+      if (typeof option === 'string') {
+        return {
+          value: option,
+          label: '',
+        };
+      }
+      return option;
+    }
+    return null;
+  }, [editingIndex, value]);
+
+  const handleOptionTextTnput = React.useCallback(
+    (
+      event:
+        | React.KeyboardEvent<HTMLInputElement | HTMLTextAreaElement>
+        | React.ChangeEvent<HTMLInputElement>,
+    ) => {
+      if ('key' in event && event.key === 'Enter') {
```

Comment:
maybe no need to check for the `key` here, since if it's not there it will just be undefined?
anyway I think this is ok too.